File: frontend/src/interfaces/Spawn.ts
export interface Spawn {
  id: number;
  strain: string;
  inoculation_type: string;
  inoculation_date: string;
  full_colonization_date: string;
  notes: string;
}

export interface CreateSpawn {
  strain: string;
  inoculation_type: string;
  inoculation_date: string;
  full_colonization_date: string;
  notes: string;
}

export interface UpdateSpawn {
  strain: Optional[string];
  inoculation_type: Optional[string];
  inoculation_date: Optional[string];
  full_colonization_date: Optional[string];
  notes: Optional[string];
}

File: backend/schemas/spawn.py
from datetime import date, datetime
from typing import Optional

from pydantic import BaseModel


class SpawnCreate(BaseModel):
    strain: str
    inoculation_type: str
    inoculation_date: Optional[datetime]
    full_colonization_date: Optional[datetime]
    notes: Optional[str]


class SpawnDB(BaseModel):
    id: int
    strain: str
    inoculation_type: str
    inoculation_date: Optional[datetime]
    full_colonization_date: Optional[datetime]
    notes: Optional[str]

    class Config:
        orm_mode = True
